src/hooks/routing/useRoutingAPITrade.ts

```diff
@@ -0,0 +1,136 @@
+import { skipToken } from '@reduxjs/toolkit/query/react'
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { useRoutingAPIArguments } from 'hooks/routing/useRoutingAPIArguments'
+import useIsValidBlock from 'hooks/useIsValidBlock'
+// import { IMetric, MetricLoggerUnit, setGlobalMetric } from '@uniswap/smart-order-router'
+// import { sendTiming } from 'components/analytics'
+import { useStablecoinAmountFromFiatValue } from 'hooks/useUSDCPrice'
+import ms from 'ms.macro'
+import { useMemo } from 'react'
+import { useGetQuoteQuery } from 'state/routing/slice'
+import { GetQuoteResult, InterfaceTrade, TradeState } from 'state/routing/types'
+import { computeRoutes, transformRoutesToTrade } from 'state/routing/utils'
+
+/**
+ * Returns the best trade by invoking the routing api or the smart order router on the client
+ * @param tradeType whether the swap is an exact in/out
+ * @param amountSpecified the exact amount to swap in/out
+ * @param otherCurrency the desired output/payment currency
+ */
+export function useRoutingAPITrade<TTradeType extends TradeType>(
+  // TODO: fxn name is misleading: uses API & SOR
+  tradeType: TTradeType,
+  routerApiBaseUrl?: string,
+  amountSpecified?: CurrencyAmount<Currency>,
+  otherCurrency?: Currency
+): {
+  state: TradeState
+  trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
+} {
+  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
+    () =>
+      tradeType === TradeType.EXACT_INPUT
+        ? [amountSpecified?.currency, otherCurrency]
+        : [otherCurrency, amountSpecified?.currency],
+    [amountSpecified, otherCurrency, tradeType]
+  )
+
+  const useClientSideRouter = !Boolean(routerApiBaseUrl) // False if URL is '' or undefined
+
+  const queryArgs = useRoutingAPIArguments({
+    tokenIn: currencyIn,
+    tokenOut: currencyOut,
+    amount: amountSpecified,
+    tradeType,
+    baseUrl: routerApiBaseUrl,
+    useClientSideRouter,
+  })
+
+  const { isLoading, isFetching, isError, data, currentData } = useGetQuoteQuery(queryArgs ?? skipToken, {
+    pollingInterval: ms`15s`,
+    refetchOnFocus: true,
+  })
+
+  const quoteResult: GetQuoteResult | undefined = useIsValidBlock(Number(data?.blockNumber) || 0) ? data : undefined
+
+  const route = useMemo(
+    () => computeRoutes(currencyIn, currencyOut, tradeType, quoteResult),
+    [currencyIn, currencyOut, quoteResult, tradeType]
+  )
+
+  // get USD gas cost of trade in active chains stablecoin amount
+  const gasUseEstimateUSD = useStablecoinAmountFromFiatValue(quoteResult?.gasUseEstimateUSD) ?? null
+
+  const isSyncing = currentData !== data
+
+  return useMemo(() => {
+    if (!currencyIn || !currencyOut) {
+      return {
+        state: TradeState.INVALID,
+        trade: undefined,
+      }
+    }
+
+    if (isFetching) {
```

Comment:
use `isFetching` instead of `isLoading && !quoteResult` to check for TradeState.LOADING

https://redux.js.org/tutorials/essentials/part-7-rtk-query-basics#using-query-hooks-in-components:~:text=isFetching%3A%20a%20boolean%20indicating%20if%20the%20hook%20is%20currently%20making%20any%20request%20to%20the%20server